core/network/besu/contracts/interop/manageAssetAny.sol

```diff
@@ -140,17 +190,29 @@ contract AssetExchangeContract is ERC1155Holder {
     }
 
     // The receiver claims the ownership of an asset locked for them once they obtain the preimage of the hashlock
-    function claimAsset(bytes32 lockContractId, bytes32 preimage)
+    function claimAsset(bytes32 lockContractId, bytes memory lockInfoProtobuf)
         external
         returns (bool)
     {
         LockContract storage c = lockContracts[lockContractId];
-
+        AssetClaimHTLC memory params;
+        (, , params) = AssetClaimHTLCCodec.decode(
+            0,
+            lockInfoProtobuf,
+            uint64(lockInfoProtobuf.length)
+        );
+        bytes32 preimage = bytesTobytes32(
+            slice(params.hashPreimageBase64, 2, 32)
+        );
         // Check the validity of the claim
         require(c.status == LOCKED, "lockContract is not active");
         require(block.timestamp < c.expirationTime, "lockContract has expired");
+        bytes32 encryptedPreimage;
```

Comment:
Its not encrypted pre-image, may be call this as computedHash? as in we are now computing hash using the preimage and comparing it with provided hash in LockContract?